tests/samplers_tests/test_samplers.py

```diff
@@ -105,11 +123,81 @@ def test_categorical(sampler_class, choices):
     distribution = CategoricalDistribution(choices)
 
     study = optuna.study.create_study(sampler=sampler_class())
+    in_trial_study = InTrialStudy(study)
     points = np.array([
-        study.sampler.sample(study.storage, study.study_id, 'x', distribution) for _ in range(100)
+        study.sampler.sample_independent(in_trial_study, new_trial(study), 'x', distribution)
+        for _ in range(100)
     ])
     # 'x' value is corresponding to an index of distribution.choices.
     assert np.all(points >= 0)
     assert np.all(points <= len(distribution.choices) - 1)
     round_points = np.round(points)
     np.testing.assert_almost_equal(round_points, points)
+
+
+def new_trial(study):
```

Comment:
I think function names usually start with verbs, but this doesn't. What do you think of the name like `create_new_trial` or `_create_new_trial` for this function?

Reply:
You're right. I renamed it at the commit https://github.com/pfnet/optuna/pull/380/commits/1ade7df8f38d650fdee3595781b001860a253daa.